iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant.inl

```diff
@@ -31,7 +31,7 @@ inline constexpr variant<Types...>::variant(const variant& rhs) noexcept
 {
     if (m_type_index != INVALID_VARIANT_INDEX)
     {
-        internal::call_at_index<0, Types...>::copyConstructor(m_type_index, rhs.m_storage, m_storage);
+        internal::call_at_index<0, Types...>::copyConstructor(m_type_index, &rhs.m_storage.data[0], &m_storage.data[0]);
```

Comment:
I think this leads to warnings with `-Wcast-align=strict`. Unfortunately this is baked into the `call_at_index` struct.

Reply:
@FerdinandSpitzschnueffler after some thinking, the `call_at_index` should use `void*` instead of `byte_t*`. Could you please change that?

With that change there is only two offender when using `-Wcast_aling=strict`

One is in `types.h` in the C binding
```cpp
/// @brief handle of the chunk header
typedef struct
{
    // could be empty but then we get `struct has no members` warning
    uint8_t do_not_touch_me[1];
} iox_chunk_header_t;
```
to
```cpp
/// @brief handle of the chunk header
typedef struct iox_chunk_header_opaque_t iox_chunk_header_t;
```

And the others in the UDS and MQ iceperf code ... which is by the way broken.

I think `types.h` and the iceperf example could be fixed in another PR.

Reply:
I changed `call_at_index` to take a `void*` and created #1778 for the warnings in the C binding and the iceperf example.

Reply:
@MatthiasKillat I like the idea of a `iox::byte` similar to `std::byte`. Would be something for a separate issue.